src/Compile-SourceScript/Public/Compile-SourceScript.ps1

```diff
@@ -123,33 +128,60 @@ function Compile-SourceScript {
-            '1' | Out-File -FilePath $stdInFile.FullName -Force -Encoding utf8
+            $stdInFile = Join-Path $SCRIPTING_DIR ".$epoch"
+            $tempDir = Join-Path ([System.IO.Path]::GetTempPath()) (New-Guid).Guid
+            $stdoutFile = Join-Path $tempDir 'stdout'
+            $stderrFile = Join-Path $tempDir 'stderr'
             $processArgs = @{
                 FilePath = $compiler.FullName
+                ArgumentList = @(
+                    if ($PSBoundParameters['SkipWrapper']) {
+                        $sourceFile.Name
+                        "-o$($MOD[$MOD_NAME]['compiled_dir_name'])/$($sourceFile.Basename)$($MOD[$MOD_NAME]['plugin_ext'])"
+                    }else {
+                        $sourceFile.Name
+                    }
+                )
                 WorkingDirectory = $SCRIPTING_DIR
-                RedirectStandardInput = $stdInFile.FullName
+                RedirectStandardInput = $stdInFile
+                RedirectStandardOutput = $stdoutFile
+                RedirectStandardError = $stderrFile
                 Wait = $true
                 NoNewWindow = $true
+                PassThru = $true
             }
-            if ($PSBoundParameters['SkipWrapper']) {
-                $processArgs['ArgumentList'] = @(
-                    $sourceFile.Name
-                    "-o$($MOD[$MOD_NAME]['compiled_dir_name'])/$($sourceFile.Basename)$($MOD[$MOD_NAME]['plugin_ext'])"
-                )
-            }else {
-                $processArgs['ArgumentList'] = @(
-                    $sourceFile.Name
-                )
+
+            # Prepare compilation environment
+            if ($item = New-Item -Path $stdInFile -ItemType File -Force) {
+                # This dummy input bypasses the 'Press any key to continue' prompt of the compiler
+                '1' | Out-File -FilePath $item.FullName -Force -Encoding utf8
             }
+            New-Item $tempDir -ItemType Directory -Force > $null
             New-Item -Path $COMPILED_DIR -ItemType Directory -Force | Out-Null
 
             # Begin compilation
+            "Compiling..." | Write-Host -ForegroundColor Cyan
             if ($PSBoundParameters['SkipWrapper']) { "Compiling $($sourceFile.Name)..." | Write-Host -ForegroundColor Yellow }
-            Start-Process @processArgs
+
+            # Compile
+            $global:LASTEXITCODE = 0
+            $p = Start-Process @processArgs
+            $stdout = Get-Content $stdoutFile
+            $stdout | Write-Host
+            $stderr = Get-Content $stderrFile
+            foreach ($line in $stdout) {
+                if ($line -match $MOD[$MOD_NAME]['compiler'][$OS]['error_regex']) {
+                    $global:LASTEXITCODE = 1
+                    break
+                }
+            }
+
+            # Cleanup
+            Remove-Item $stdInFile -Force
+            Remove-Item $tempDir -Recurse -Force
```

Comment:
These would not execute if the compile process terminates prematurely.

Reply:
Should not have a terminating at compile, or else use a `try catch` block and `finally` here to cleanup.

Reply:
Yes, a nested try-catch-finally block should work.

Reply:
or we could ensure `ErrorActionPreference = ‘Continue’` for that part

Reply:
Which part?

Reply:
when preparing and compiling and cleaning up. same block of code try catch would wrap around. @joeltimothyoh

Reply:
Fixed.